src/ocamlorg_package/lib/opam_repository.ml

```diff
@@ -72,23 +72,38 @@ let last_commit () =
   in
   output
 
-let ls_dir directory =
+let fold_dir f acc directory =
   match Sys.readdir directory with
   | exception Sys_error _ ->
     None
   | entries ->
-    let entry_is_dir x = Sys.is_directory (Filename.concat directory x) in
-    Some (List.filter entry_is_dir (Array.to_list entries))
-
-let list_packages () =
-  match ls_dir Fpath.(to_string (clone_path / "packages")) with
-  | Some pkgs ->
-    pkgs
-  | None ->
-    []
+    (* Sort to remove a source of non-determinism. *)
+    Array.sort String.compare entries;
```

Comment:
This is basically useless because these values are either inserted in a `Map` or thrown away immediately. But I really don't like leaving this unsorted.